bundles/org.openhab.ui.basic/src/main/resources/snippets/input.html

```diff
@@ -0,0 +1,26 @@
+<div class="mdl-form__row mdl-cell mdl-cell--6-col mdl-cell--8-col-tablet %visibility_class%">
+	<span class="mdl-form__icon">
+		<img data-icon="%category%" src="../icon/%category%?state=%state%&format=%icon_type%&anyFormat=true" />
+	</span>
+	<span %labelstyle% class="mdl-form__label">
+		%label%
+	</span>
+	<div
+		class="mdl-form__control mdl-form__input mdl-textfield mdl-js-textfield"
+		data-control-type="input"
+		data-item="%item%"
+		data-widget-id="%widget_id%"
+		data-item-type=%data_type%
+		for="oh-input-%item%"
+	>
+		<input
+			%valuestyle% type="text"
+			id="oh-input-%item%"
+			class="mdl-textfield__input"
+			value="%data_state%"
+		/>
+		<label for="oh-input-%item%" class="mdl-textfield__label">
```

Comment:
That would do it. But I am not convinced we should. You cannot set NULL or UNDEF through the UI. If you set anything, it would always obey the color rules. I on purpose refrained from setting a color for NULL or UNDEF to make it very apparent that the - means no value is set. The MDL input element also on purpose uses a different color for the placeholder, and the - is just a placeholder. The color of the placeholder is the same as the borderline color. I tweaked these to be visible, but not too strong on light and dark themes.

Reply:
NULL/UNDEF can be set by the binding itself.

My idea is only to be consistent with all other existing widgets, starting with the Text widget.
When the user defines a default color (through the valuecolor attribute), this color will be used when the item state is NULL or UNDEF.
So I am asking myself whether we should introduced something different here.

Reply:
What I try to explain is that until now the user has a full control on label/value colors (depending on item(s) state). I think you are introducing a case where value color is hardcoded. Are we OK on that point ?

Reply:
Maybe your option with hardcoded color is the best option in that case, I am just not sure.